media_driver/agnostic/common/vp/hal/vphal_render_blitter.cpp

```diff
@@ -0,0 +1,124 @@
+	):RenderState(pOsInterface, pRenderHal, pPerfData, peStatus)
+{}
+
+
+VPHAL_BLITTER_STATE::~VPHAL_BLITTER_STATE()
+{}
+
+int VPHAL_BLITTER_STATE::iframe=0;
+
+MOS_STATUS VPHAL_BLITTER_STATE::Initialize(
+        const VphalSettings    *pSettings,
+        Kdll_State             *pKernelDllState)
+{
+	MOS_STATUS                       eStatus;
+	PRENDERHAL_INTERFACE             pRenderHal;
+	PVPHAL_BLITTER_STATE              pBlitterState = this;
+
+    eStatus = MOS_STATUS_SUCCESS;
+	pRenderHal	 = pBlitterState->m_pRenderHal;
+
+	if (pRenderHal == nullptr)
+    {
+        eStatus = MOS_STATUS_UNKNOWN;
+        goto finish;
+    }
+
+	MOS_UNUSED(pSettings);
+    VPHAL_RENDER_CHK_NULL(pKernelDllState);
+
+	if (m_reporting == nullptr)
+    {
+        m_reporting = MOS_New(VphalFeatureReport);
+    }
+
+	pBlitterState->m_pKernelDllState = pKernelDllState;
+
+finish:
+	return eStatus;
+
+}
+
+MOS_STATUS VPHAL_BLITTER_STATE::Render(
+    PCVPHAL_RENDER_PARAMS  pcRenderParams,
+    RenderpassData         *pRenderPassData)         // RenderpassData isn't used
+{
+	MOS_STATUS              eStatus = MOS_STATUS_SUCCESS;
+
+	PVPHAL_SURFACE           pSrcSurface;
+    PVPHAL_SURFACE           pOutputSurface;
+    pSrcSurface    = pcRenderParams->pSrc[0];
+    pOutputSurface = pcRenderParams->pTarget[0];
+    PMOS_INTERFACE           pOsInterface;
+	PVPHAL_BLITTER_STATE     pBlitterState = this;
+    pOsInterface            = pBlitterState->m_pOsInterface;
+
+
+
+	MOS_UNUSED(pcRenderParams);
+	VPHAL_RENDER_ASSERT(pSrcSurface);
+    VPHAL_RENDER_ASSERT(pOutputSurface);
+
+	drm_intel_bo_switch(pSrcSurface->OsResource.bo , pOutputSurface->OsResource.bo,  pSrcSurface->OsResource.pGmmResInfo->GetBaseHeight() * 3 / 2, pSrcSurface->OsResource.pGmmResInfo->GetRenderPitch());
+
+    if (iframe == 0)
+	{
+		usleep(10000);   //sleep 10ms for the first frame
+		iframe++;
+	}
+    VpHal_RndrUpdateStatusTableAfterSubmit(pOsInterface, &m_StatusTableUpdateParams, MOS_GPU_CONTEXT_BLITTER, eStatus);
+    return eStatus;
+}
+
+MOS_STATUS VpHal_RndrRenderBlitter(
+    VphalRenderer           *pRenderer,
+    PVPHAL_RENDER_PARAMS   pRenderParams,
+    RenderpassData          *pRenderPassData)
+{
+	MOS_STATUS               eStatus;
+    PMOS_INTERFACE           pOsInterface;
+    RenderState              *pRenderState;
+    VphalFeatureReport*      pReport;
+
+
+    //------------------------------------------------------
+    VPHAL_RENDER_ASSERT(pRenderer);
+    VPHAL_RENDER_ASSERT(pRenderParams);
+    //------------------------------------------------------
+
+    eStatus                 = MOS_STATUS_SUCCESS;
+    pReport                 = pRenderer->GetReport();
+    pRenderState            = pRenderer->pRender[VPHAL_RENDER_ID_BLITTER];
+
+
+
+    VPHAL_RENDER_CHK_NULL(pRenderState);
+    VPHAL_RENDER_ASSERT(pRenderState->GetRenderHalInterface());
+
+	pRenderer->pRender[VPHAL_RENDER_ID_VEBOX+pRenderer->uiCurrentChannel]->SetStatusReportParams(pRenderer, pRenderParams);
+
```

Comment:
Which place to initialize the "uiCurrentChannel"? Suggest to use VPHAL_RENDER_ID_BLITTER to find the render instance.